### PR TITLE
Describe release process

### LINKED ISSUE #375: Describe release process
Currently, the repository is missing a readme file describing release process.

### PR DESCRIPTION
**Pull request by yurishkuro**
Resolves #375

### CODE REVIEW

**Comment by yurishkuro** [comment]
cc @pavolloffay

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Jaeger Backend Release Process
+
+1. Create a PR "Preparing release X.Y.Z" against master branch (e.g. #351). It must include:
```
Comment (pavolloffay) [inline RELEASE.md]: link #351 does not work

Comment (pavolloffay) [reply RELEASE.md]: The text does not mention `changelog.md` file.

Comment (yurishkuro) [reply RELEASE.md]: the link works for me - what do you get?

Comment (pavolloffay) [reply RELEASE.md]: When I hit the view it does now work https://github.com/jaegertracing/jaeger/blob/1a684a8a04e6e8ddd02026760cbe52257a9e6830/RELEASE.md

**Pull request merged**